packages/components/inputs/selectable-search-input/src/selectable-search-input.tsx

```diff
@@ -0,0 +1,446 @@
+          name: getTextInputName(props.name),
+          value: event.target.value,
+        },
+      });
+    }
+  };
+
+  const handleSubmit = (
+    event: KeyboardEvent<HTMLButtonElement> | MouseEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+    if (props.onSubmit) {
+      props.onSubmit({
+        textValue: searchValue,
+        dropdownValue: selectedOption?.value,
+      });
+    }
+  };
+
+  const dropdownName = getDropdownName(props.name);
+  const dropdownId = SelectableSearchInput.getDropdownId(
+    selectablSearchInputId
+  );
+
+  const handleDropdownFocus = useCallback(() => {
+    if (props.onFocus) {
+      props.onFocus({
+        target: {
+          id: dropdownId,
+          name: dropdownName,
+        },
+      });
+    }
+    toggleDropdownHasFocus(true);
+  }, [props.onFocus, toggleDropdownHasFocus, dropdownName, dropdownId]);
+
+  const handleDropdownBlur = useCallback(() => {
+    toggleDropdownHasFocus(false);
+  }, [toggleDropdownHasFocus]);
+
+  const handleContainerBlur = useCallback(
+    (event) => {
+      // ensures that both fields are marked as touched when one of them
+      // is blurred
+      if (
+        typeof props.onBlur === 'function' &&
+        !containerRef.current?.contains(event.relatedTarget)
+      ) {
+        props.onBlur({
+          target: {
+            id: SelectableSearchInput.getDropdownId(selectablSearchInputId),
+            name: getDropdownName(props.name),
+          },
+        });
+        props.onBlur({
+          target: {
+            id: SelectableSearchInput.getTextInputId(selectablSearchInputId),
+            name: getTextInputName(props.name),
+          },
+        });
+      }
+    },
+    [props.onBlur, selectablSearchInputId, props.name]
+  );
+
+  return (
+    <Constraints.Horizontal max={props.horizontalConstraint}>
+      <Container
+        ref={containerRef}
+        onBlur={handleContainerBlur}
+        data-testid="selectable-search-input-container"
+      >
+        <Constraints.Horizontal max={props.menuHorizontalConstraint}>
+          <SelectableSelect
+            {...props}
+            id={SelectableSearchInput.getDropdownId(selectablSearchInputId)}
+            name={getDropdownName(props.name)}
+            textInputHasFocus={textInputHasFocus}
+            dropdownHasFocus={dropdownHasFocus}
+            handleDropdownFocus={handleDropdownFocus}
+            handleDropdownBlur={handleDropdownBlur}
+            isNewTheme={isNewTheme}
+            textInputRef={textInputRef}
+            selectedOption={selectedOption}
+          />
+        </Constraints.Horizontal>
+        <div
+          css={[
+            getSelectableSearchInputContainerStyles(props),
+            dropdownHasFocus &&
+              isNewTheme &&
+              css`
+                border-left-color: ${designTokens.borderColorForInputWhenFocused};
+              `,
+          ]}
+        >
+          <input
```

Comment:
Might be good to also support submitting on <kbd>Enter</kbd>

Reply:
Don't we get that it we wrap the input into a form nicely?